### PR TITLE
Tk245 - Add new subscribtion functionaliy from Batcave

### PR DESCRIPTION
**Pull request by karuna835**
1. Drop down to select program & Batch (Same as how we do Swap)
2. Adding Subs & Subs_meta entry to backend

### CODE REVIEW

File: app/apis/btcapi/subs/save.php [hunk 1]
```diff
@@ -0,0 +1,45 @@
+<?php
+//Tk245 - Adding Subs & Subs_meta entry to backend.
```
Comment (kailasb) [inline app/apis/btcapi/subs/save.php:2]: add authentication line

File: app/apis/btcapi/subs/save.php [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+
+	$subs["combo"] = "";
+	if(!empty($get_combo))
+	{
+		$subs["combo"] = $get_combo[0]['combo'];
+	}
+	
+	if(!empty($subs))
+	{
+		subs_create($subs["user_id"], $subs);
```
Comment (kailasb) [inline app/apis/btcapi/subs/save.php:26]: get subs id then update on that id

File: app/apis/btcapi/subs/save.php [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+
+	}
+	
+    $subs["batch_id"] = db_sanitize($data->batch_id);
+	$subs["bundle_id"] = db_sanitize($data->course_id);
+	$subs["agent_id"] = $_SESSION["user"]["user_id"];
+	$subs["status"] = "pending";
+	$subs["user_id"] = db_sanitize($data->user_id);
+	$get_combo = db_query("SELECT combo FROM course_bundle WHERE bundle_id = ".$subs["bundle_id"].";");
+	$get_corp = db_query("SELECT corp FROM subs WHERE user_id = ".$subs["user_id"]." AND corp != '';");
```
Comment (kailasb) [inline app/apis/btcapi/subs/save.php:16]: add limit in as you are using only one record. so there is no use of getting all records from database fro that user

File: app/apis/btcapi/subs/save.php [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+				$corp = $get_corp[0]['corp'];
+				
+				db_exec("UPDATE subs SET corp = '".$corp."' WHERE subs_id = ".$sub_id.";");
+			}
+		}
+		$response = ["status" => true, "msg" => ""];
+	    die(json_encode($response));
+
+	}
+
```
Comment (kailasb) [inline app/apis/btcapi/subs/save.php:43]: return message if non of the condition is satisfy

File: b/js/app.js [hunk 5]
```diff
@@ -1163,9 +1163,12 @@ angular.module("batcave", ["ngRoute"])
 		$scope.bundleChange = function() {
 			if ($scope.editableSubs.bundle_id != $scope.editableSubs.originalBundleId) {
 				$scope.editableSubs.batch = "0";
-				if ($scope.editableSubs.enr.length>0) {
-					$scope.editableSubsEnrs = $scope.editableSubs.enr;
-					$scope.editableSubs.enr = [];
+				//Tk245 - Added Check for $scope.editableSubs.enr undefined or not
+				if($scope.editableSubs.enr != undefined){
```
Comment (kailasb) [inline b/js/app.js:1167]: undefined is not a variable

File: b/views/user.html [hunk 6]
```diff
@@ -697,7 +697,8 @@ <h5 class="modal-title">{{editableSubs.editable ? 'Edit' : 'View'}} subscription
 		</div>
 	</div>
 </div>
-<div class="modal fade" id="profile" tabindex="-1" role="dialog">
+<!-- TK245 - set tabinex empty -->
+<div class="modal fade" id="profile" tabindex="" role="dialog">
```
Comment (kailasb) [inline b/views/user.html:701]: why tab index is removed

File: b/js/app.js [hunk 7]
```diff
@@ -1345,6 +1348,18 @@ angular.module("batcave", ["ngRoute"])
 			}
 			angular.element("#exportInfo").modal("show");
 		}
+		//Tk245 - Write function for save new subscription
+		$scope.saveNewSubs = function() {
+			$http.post(API + "/subs/save", {batch_id: $scope.editableSubs.batch,
```
Comment (kailasb) [inline b/js/app.js:1353]: check this function is trigger for which functionality. as the save button is common in the popup and there is other tasbs in the popup

**Review by kailasb — COMMENTED** [review]
Resolved the all the changes then commit once gain.

File: app/apis/btcapi/subs/save.php (hunk 1 above)
Comment (karuna835) [reply app/apis/btcapi/subs/save.php:4]: Added authentication line

File: app/apis/btcapi/subs/save.php (hunk 2 above)
Comment (karuna835) [reply app/apis/btcapi/subs/save.php]: Updated on subs id

File: app/apis/btcapi/subs/save.php (hunk 4 above)
Comment (karuna835) [reply app/apis/btcapi/subs/save.php:45]: added return message

File: b/js/app.js (hunk 5 above)
Comment (karuna835) [reply b/js/app.js]: I have checked this condition for dynamic used some time variable get undefined

File: b/views/user.html (hunk 6 above)
Comment (karuna835) [reply b/views/user.html:701]: select2 search not working because of that I removed.

File: b/js/app.js (hunk 7 above)
Comment (karuna835) [reply b/js/app.js]: Resolved this. now data saved on a particular tab only

File: app/apis/btcapi/subs/save.php [hunk 8]
```diff
@@ -23,12 +25,12 @@
 	
 	if(!empty($subs))
 	{
-		subs_create($subs["user_id"], $subs);
-		$get_last_record = db_query("SELECT subs_id FROM subs ORDER BY subs_id DESC 
-   LIMIT 1;");
-        if(!empty($get_last_record))
+		$sub_data = subs_create($subs["user_id"], $subs);
+		//$get_last_record = db_query("SELECT subs_id FROM subs ORDER BY subs_id DESC
```
Comment (kailasb) [inline app/apis/btcapi/subs/save.php]: removed unused/commented code

File: app/apis/btcapi/subs/save.php [hunk 9]
```diff
@@ -38,8 +40,8 @@
 		}
 		$response = ["status" => true, "msg" => ""];
 	    die(json_encode($response));
```
Comment (kailasb) [inline app/apis/btcapi/subs/save.php]: Optimize code

**Review by kailasb — CHANGES_REQUESTED** [review]

File: app/apis/btcapi/subs/save.php (hunk 8 above)
Comment (karuna835) [reply app/apis/btcapi/subs/save.php]: Remove comment line

File: app/apis/btcapi/subs/save.php (hunk 9 above)
Comment (karuna835) [reply app/apis/btcapi/subs/save.php]: done

**Review by kailasb — APPROVED** [review]
code is proper

**Pull request merged**